src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -767,58 +831,203 @@ public IEnumerable<string> GetCollections(Transaction transaction)
+
+                        if (config.MinimumRevisionAgeToKeep.HasValue)
+                        {
+                            var dbNow = _database.Time.GetUtcNow();
+                            var revLastModify = revision.LastModified;
+                            var dif = dbNow - revLastModify;
+                            if (dif <= config.MinimumRevisionAgeToKeep.Value)
                             {
-                                currentCollection = docCollection;
-                                writeTable = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, new CollectionName(docCollection));
+                                reachedNotOutOfDate = true;
+                                continue;
                             }
+                        }
 
-                            writeTable.DeleteByKey(keySlice);
+                        conflictDeletedCount++;
+                    }
+                    else
+                    {
+                        if (finishedRegualr)
+                        {
+                            continue;
                         }
+
+                        if (revision.Flags.Contain(DocumentFlags.ForceCreated) && skipForceCreated)
+                        {
+                            skippedForceCreated++;
+                            continue;
+                        }
+                        regularDeletedCount++;
                     }
 
-                    deletedRevisionsCount++;
-                    break;
+                    var collection = CollectionName.GetCollectionName(revision.Data);
+                    revisionsToRemove.Add((revision.ChangeVector, revision.Etag, revision.Flags, collection));
                 }
 
-                if (hasValue == false)
-                    break;
             }
 
+            var maxEtagDeleted = DeleteRevisionsInternal(context, table, collectionName, changeVector, lastModifiedTicks, revisionsToRemove);
+
+            _database.DocumentsStorage.EnsureLastEtagIsPersisted(context, maxEtagDeleted);
+            return regularDeletedCount + conflictDeletedCount;
+        }
+
+        private long GetConflictRevisionsCount(
+            DocumentsOperationContext context, Table table, Slice prefixSlice)
+        {
+            long count = 0;
+            foreach (var read in table.SeekForwardFrom(RevisionsSchema.Indexes[IdAndEtagSlice], prefixSlice, skip: 0, startsWith: true))
+            {
+                var tvr = read.Result.Reader;
+                using (var revision = TableValueToRevision(context, ref tvr))
+                {
+                    if (revision.Flags.Contain(DocumentFlags.Conflicted) || revision.Flags.Contain(DocumentFlags.Resolved))
+                        count++;
+                }
+            }
+
+            return count;
+        }
+
+        private long DeleteForceCreatedRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
```

Comment:
I think this isn't what we intended to have

Reply:
I don't think we want to have something like `deletes all the force-created revisions`

Reply:
ok